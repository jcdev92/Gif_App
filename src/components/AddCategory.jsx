import PropTypes from "prop-types";

export function AddCategory({ onAddCategory }) {

  const submitCategory = (e) => {
    e.preventDefault();

    const inputValue = e.target.category.value;
    const inputCategory = inputValue.trim();

    if (inputCategory.length <= 1) return;
    onAddCategory(inputCategory);

    // clear input
    e.target.category.value = "";
  };

  return (
    <div>
      <h1>GifExpertApp</h1>
      <form onSubmit={submitCategory} aria-label="form">
        <input type="text" name="category"/>
        <button type="submit">Add</button>
      </form>
    </div>
  );
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired,
}