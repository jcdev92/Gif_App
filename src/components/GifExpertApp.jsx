import React, { useState } from "react";
console.log("".length);

function GifExpertApp(props) {
  const [categories, setCategories] = useState([
    "One Punch",
    "Samurai X",
    "Dragon Ball",
  ]);

  const submitCategory = (e) => {
    e.preventDefault();
    setCategories([...categories, e.target.category.value]);
    // clear input
    e.target.category.value = "";
    // reset state
  };

  return (
    <div>
      <h1>GifExpertApp</h1>
      <form onSubmit={submitCategory}>
        <input type="text" name="category" />
        <button type="submit">Add</button>
      </form>
      <ol>
        {categories.map((category) => {
          const randomKey =
            Math.random().toString(36).substring(2, 15) +
            Math.random().toString(36).substring(2, 15);
          return <li key={randomKey}>{category}</li>;
        })}
      </ol>
    </div>
  );
}

export default GifExpertApp;
