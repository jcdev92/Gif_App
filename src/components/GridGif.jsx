import React , { useEffect , useState } from "react";
import { getGifs }                      from "../helpers/getGif.js";

export const GridGif = ({ category, key }) => {
  const [images, setImages] = useState([]);

  const getImages = async () => {
    const images = await getGifs(category);
    setImages(images);
  };

  useEffect(() => {
    getImages();
  }, []);

  return (
    <>
      <h2>{category}</h2>
      <div className="card-grid">
        {images.map(({id, title, url}) => (
          <div key={id} className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
          </div>
        ))}
      </div>
    </>
  );
};
