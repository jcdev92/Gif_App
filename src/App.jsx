import './App.css'
import GifExpertApp from "./components/GifExpertApp.jsx";

function App() {

  return (
    <div className="App">
        <GifExpertApp />
    </div>
  )
}

export default App
