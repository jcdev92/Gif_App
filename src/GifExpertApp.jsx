import "./App.css";
import React, { useState } from "react";
import { AddCategory, GridGif } from "./components";

export function GifExpertApp() {
  const [categories, setCategories] = useState(["One Punch"]);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  };

  return (
    <div className="App">
      <AddCategory onAddCategory={onAddCategory} />
      <hr />
      {categories.map((category) => (
        <GridGif key={category} category={category} />
      ))}
    </div>
  );
}
