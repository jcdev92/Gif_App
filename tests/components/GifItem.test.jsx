import { render }  from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('Test on <GifItem />', () => {
    const title = 'A Title';
    const url = 'https://localhost/something.jpg';
    test('It should match with the snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

});