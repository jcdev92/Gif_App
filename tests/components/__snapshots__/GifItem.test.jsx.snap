// Vitest Snapshot v1

exports[`Test on <GifItem /> > It should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="A Title"
      src="https://localhost/something.jpg"
    />
    <p>
      A Title
    </p>
  </div>
</div>
`;
